src/Race.js

```diff
@@ -486,33 +490,38 @@ export default class Race {
 			}
 		} else {
 			// close down race if this is the last person leaving
-			this.channel.race = new Race(this.channel, this.game);
+			this.closeRace();
 			note = " Closing race.";
 		}
 
 		return `${entrant.cleanName} left the race.${note}`;
 	}
 
 	/**
-	 * Resets all race entrants
+	 * Resets the race
 	 */
-	resetEntrants() {
-		for (let entrant of this.entrantIterator()) {
-			this.resetEntrant(entrant);
+	closeRace() {
+		this.channel.race = new Race(this.channel, this.game);
+		for (let entrant of this.allEntrantsEver) {
+			this.resetEntrant(entrant, true);
 		}
```

Comment:
That doesn't always work, unfortunately. The player might already be in a different IL race in a different channel on the same server. In that case it would be surprising to see your score suddenly drop to 0 in the middle of an IL series :D

I know it's just an unlikely edge case, but I'd rather not structure the code like this. I'm gonna think about it a bit.